### PR TITLE
feat: `validateParameters`

### PR DESCRIPTION
**Pull request by 0xmoebius**
# 🤖 Linear

Closes OPO-608

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,8 @@
     "package.json": "sort-package-json"
   },
   "dependencies": {
-    "@defi-wonderland/prophet-core-contracts": "0.0.0-49fc8fa6",
+    "@defi-wonderland/prophet-core-contracts": "0.0.0-6ac0f50d",
+    "@defi-wonderland/solidity-utils": "0.0.0-3e9c8e8b",
```
Comment (gas1cent) [inline package.json:42]: Why here, not dev dependency?

File: solidity/contracts/modules/dispute/BondEscalationModule.sol [hunk 2]
```diff
@@ -295,4 +295,19 @@ contract BondEscalationModule is Module, IBondEscalationModule {
...
+    pure
+    override(Module, IModule)
+    returns (bool _valid)
+  {
+    RequestParameters memory _params = decodeRequestData(_encodedParameters);
+    _valid = (
+      address(_params.accountingExtension) == address(0) || address(_params.bondToken) == address(0)
+        || _params.bondSize == 0 || _params.bondEscalationDeadline == 0 || _params.maxNumberOfEscalations == 0
+        || _params.tyingBuffer == 0 || _params.disputeWindow == 0
+    ) ? false : true;
```
Comment (gas1cent) [inline solidity/contracts/modules/dispute/BondEscalationModule.sol:311]: I'd refactor this a bit, for instance:
```suggestion
_valid = address(_params.accountingExtension) != address(0) &&
      address(_params.bondToken) != address(0) &&
      _params.bondSize > 0 &&
      _params.bondEscalationDeadline > 0 &&
      _params.maxNumberOfEscalations > 0 &&
      _params.tyingBuffer > 0 &&
      _params.disputeWindow > 0;
```

File: package.json (hunk 1 above)
Comment (0xmoebius) [reply package.json:42]: it has always been in `dependencies` iirc

File: solidity/contracts/modules/dispute/BondEscalationModule.sol (hunk 2 above)
Comment (0xmoebius) [reply solidity/contracts/modules/dispute/BondEscalationModule.sol:311]: oops. formatting gone wrong

File: solidity/test/unit/modules/dispute/BondEscalationModule.t.sol [hunk 3]
```diff
@@ -183,6 +183,21 @@ contract BondEscalationModule_Unit_ModuleData is BaseTest {
...
+    if (
+      address(_params.accountingExtension) == address(0) || address(_params.bondToken) == address(0)
+        || _params.bondSize == 0 || _params.bondEscalationDeadline == 0 || _params.maxNumberOfEscalations == 0
+        || _params.tyingBuffer == 0 || _params.disputeWindow == 0
+    ) {
+      assertFalse(bondEscalationModule.validateParameters(abi.encode(_params)));
+    } else {
+      assertTrue(bondEscalationModule.validateParameters(abi.encode(_params)));
+    }
+  }
```
Comment (gas1cent) [inline solidity/test/unit/modules/dispute/BondEscalationModule.t.sol:200]: What do you think about this testing approach?

- Declare completely valid parameters (maybe fuzz + vm.assume to make sure every value is acceptable)
- Assert that the validation passes
- Set one value to 0, assert that the validation fails
- Set all to 0, assert again

File: solidity/test/unit/modules/dispute/BondedDisputeModule.t.sol [hunk 4]
```diff
@@ -76,9 +76,23 @@ contract BondedResponseModule_Unit_ModuleData is BaseTest {
...
+    ) {
+      assertFalse(bondedDisputeModule.validateParameters(abi.encode(_params)));
+    } else {
+      assertTrue(bondedDisputeModule.validateParameters(abi.encode(_params)));
+    }
+  }
 }
 
-contract BondedResponseModule_Unit_OnDisputeStatusChange is BaseTest {
+contract BondedDisputeModule_Unit_OnDisputeStatusChange is BaseTest {
```
Comment (gas1cent) [inline solidity/test/unit/modules/dispute/BondedDisputeModule.t.sol:95]: 👏🏻👏🏻👏🏻

**Review by gas1cent — APPROVED** [review]
🎩🪄

**Pull request merged**